File: messageGo/grpc/pubsubEventData/global_log.go
package pubsubEventData

import (
	base_data "game-message-core/grpc/baseData"
	"game-message-core/proto"
)

// 日志采集
type GlobalLogEvent struct {
	MsgVersion   int64                 `json:"msgVersion"` // 消息版本号 值为毫秒时间戳
	FormService  base_data.ServiceData `json:"formService"`
	Action       proto.GameLogType     `json:"action"`
	UserId       int64                 `json:"userId"`
	ObjectType   string                `json:"objectType"`
	ObjectId     int64                 `json:"objectId"`
	ObjectName   string                `json:"objectName"`
	ObjectOwner  int64                 `json:"objectOwner"`
	ObjectCid    uint32                `json:"objectCid"`
	ObjectNumber int32                 `json:"objectNumber"`
	Pos          base_data.GrpcVector3 `json:"pos"`
	Desc         string                `json:"desc"`
}

func (p *GlobalLogEvent) Clear() {
	p.MsgVersion = 0
	p.FormService.Clear()
	p.Action = 0
	p.UserId = 0
	p.ObjectType = ""
	p.ObjectId = 0
	p.ObjectName = ""
	p.ObjectOwner = 0
	p.ObjectCid = 0
	p.ObjectNumber = 0
	p.Pos.Clear()
	p.Desc = ""
}

File: messageDotNet/grpc/EventData/GlobalLog.cs
using System;
using UnityEngine;


/// <summary>
/// dapr call 日志采集
/// </summary>
[Serializable]
public class GlobalLogEvent
{
    // 消息版本号 值为毫秒时间戳
    public long MsgVersion;
    public ServiceData FormService;
    public GameMessageCore.GameLogType Action;
    public long UserId;
    public string ObjectType;
    public long ObjectId;
    public string ObjectName;
    public int64 ObjectOwner;
    public uint ObjectCid;
    public int ObjectNumber;
    public GrpcVector3 Pos;
    public string Desc;
}
